src/ast/codegen_llvm.cpp

```diff
@@ -589,11 +589,15 @@ void CodegenLLVM::visit(Call &call)
 
     auto scoped_del = accept(call.vargs->front());
     auto arg0 = call.vargs->front();
+    // arg0 is already on the bpf stack -> use probe kernel
+    // otherwise ->  addrspace of arg0->type
+    // case : 'struct MyStruct { char b[4]; };
+    // $s = (struct MyStruct *)arg0; buf($s->b, 4)
```

Comment:
We should get rid of the probe read here, doing it for onstack data is wrong (out of scope here tho)

Reply:
Right